lib/dispatch-rider/scheduled_job.rb

```diff
@@ -0,0 +1,59 @@
+require "active_record"
+
+# @note: Later this could be pulled out to its own gem and included depending on what ORM the user
+# prefers. Something like:
+# @example
+#  gem "dispatch_rider-active_record"
+#  gem "dispatch_rider-rom" # Ruby Object Mapper
+#  gem "dispatch_rider-mongo_mapper"
+#  gem "dispatch_rider-ohm"
+module DispatchRider
+  class ScheduledJob < ActiveRecord::Base
+    class << self
+      def publisher
+        @publisher ||= Publisher.new
+      end
+
+      # @param [ActiveSupport::Duration] every
+      # @example: DispatchRider::ScheduledJob.publish_due_jobs every 1.minute
+      def publish_due_jobs(every: nil)
+        loop {
+          claim_stub = get_new_claim_stub
+          due.unclaimed.update_all claim_stub
```

Comment:
this still doesn't solve contention .. since both can do a brute forced update_all like that .. 

the only true atomic would be to fetch the record first, assume that is its current state, your claiming would assert the current claim state .. eg if its nil .. you would do .where(claim_id: nil, claim_expires_at: nil) .. actually that would be enough .. you would get the first item, claim it with a where .. 

```ruby
due.unclaimed.each do |item|
  due.where(key_attributes_from(item)).update(claim_stub)
end
claimed_items = due.where(claim_stub)
```

does this logic make sense?